### PR TITLE
Improved augroup

### PR DESCRIPTION
**Pull request by osyo-manga**
Used `augroup` and `autocmd Filetype ruby`.

### CODE REVIEW

File: plugin/ruby_hl_lvar.vim [hunk 1]
```diff
@@ -14,31 +14,27 @@ let g:ruby_hl_lvar_show_warnings =
...
+	endif
+
+	call ruby_hl_lvar#refresh(1)
+	augroup ruby_hl_lvar
+		autocmd! * <buffer>
+		autocmd TextChanged <buffer> call ruby_hl_lvar#refresh(0)
+		autocmd InsertEnter <buffer> call ruby_hl_lvar#disable(0)
+		autocmd InsertLeave <buffer> call ruby_hl_lvar#refresh(0)
 	augroup END
+	return
```
Comment (pocke) [inline plugin/ruby_hl_lvar.vim]: Why is the `return` necessary? It looks unnecessary.

File: plugin/ruby_hl_lvar.vim [hunk 2]
```diff
@@ -14,31 +14,27 @@ let g:ruby_hl_lvar_show_warnings =
 
 augroup ruby_hl_lvar
 	autocmd!
-	autocmd! Filetype * call Ruby_hl_lvar_filetype()
+	autocmd Filetype ruby call Ruby_hl_lvar_filetype()
```
Comment (pocke) [inline plugin/ruby_hl_lvar.vim]: I guess it does not work on `ruby.rspec` filetype.

**Comment by pocke** [comment]
I'm not sure what you improved. Can you tell me Pros (and Cons)?

File: plugin/ruby_hl_lvar.vim (hunk 1 above)
Comment (osyo-manga) [reply plugin/ruby_hl_lvar.vim]: Thanks, removed it.

File: plugin/ruby_hl_lvar.vim (hunk 2 above)
Comment (osyo-manga) [reply plugin/ruby_hl_lvar.vim]: oops, Thanks.

**Comment by pocke** [comment]
@osyo-manga told me the Pros and Cons in Twitter.
First, we can reset the autocmds if the autocmds are wrapped in augroup.
Second, `autocmd <buffer>` is defined as buffer local autocmd. So the dynamic group name is no longer necessary.

**Pull request merged**